// @aws-sdk/client-auto-scaling/dist/es/models/models_0.js
import { __assign } from "tslib";
export var ActiveInstanceRefreshNotFoundFault;
(function (ActiveInstanceRefreshNotFoundFault) {
    ActiveInstanceRefreshNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ActiveInstanceRefreshNotFoundFault || (ActiveInstanceRefreshNotFoundFault = {}));
export var ScalingActivityStatusCode;
(function (ScalingActivityStatusCode) {
    ScalingActivityStatusCode["Cancelled"] = "Cancelled";
    ScalingActivityStatusCode["Failed"] = "Failed";
    ScalingActivityStatusCode["InProgress"] = "InProgress";
    ScalingActivityStatusCode["MidLifecycleAction"] = "MidLifecycleAction";
    ScalingActivityStatusCode["PendingSpotBidPlacement"] = "PendingSpotBidPlacement";
    ScalingActivityStatusCode["PreInService"] = "PreInService";
    ScalingActivityStatusCode["Successful"] = "Successful";
    ScalingActivityStatusCode["WaitingForELBConnectionDraining"] = "WaitingForELBConnectionDraining";
    ScalingActivityStatusCode["WaitingForInstanceId"] = "WaitingForInstanceId";
    ScalingActivityStatusCode["WaitingForInstanceWarmup"] = "WaitingForInstanceWarmup";
    ScalingActivityStatusCode["WaitingForSpotInstanceId"] = "WaitingForSpotInstanceId";
    ScalingActivityStatusCode["WaitingForSpotInstanceRequestId"] = "WaitingForSpotInstanceRequestId";
})(ScalingActivityStatusCode || (ScalingActivityStatusCode = {}));
export var Activity;
(function (Activity) {
    Activity.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Activity || (Activity = {}));
export var ActivitiesType;
(function (ActivitiesType) {
    ActivitiesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ActivitiesType || (ActivitiesType = {}));
export var ActivityType;
(function (ActivityType) {
    ActivityType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ActivityType || (ActivityType = {}));
export var AdjustmentType;
(function (AdjustmentType) {
    AdjustmentType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AdjustmentType || (AdjustmentType = {}));
export var Alarm;
(function (Alarm) {
    Alarm.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Alarm || (Alarm = {}));
export var AlreadyExistsFault;
(function (AlreadyExistsFault) {
    AlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AlreadyExistsFault || (AlreadyExistsFault = {}));
export var AttachInstancesQuery;
(function (AttachInstancesQuery) {
    AttachInstancesQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttachInstancesQuery || (AttachInstancesQuery = {}));
export var ResourceContentionFault;
(function (ResourceContentionFault) {
    ResourceContentionFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceContentionFault || (ResourceContentionFault = {}));
export var ServiceLinkedRoleFailure;
(function (ServiceLinkedRoleFailure) {
    ServiceLinkedRoleFailure.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ServiceLinkedRoleFailure || (ServiceLinkedRoleFailure = {}));
export var AttachLoadBalancersResultType;
(function (AttachLoadBalancersResultType) {
    AttachLoadBalancersResultType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttachLoadBalancersResultType || (AttachLoadBalancersResultType = {}));
export var AttachLoadBalancersType;
(function (AttachLoadBalancersType) {
    AttachLoadBalancersType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttachLoadBalancersType || (AttachLoadBalancersType = {}));
export var AttachLoadBalancerTargetGroupsResultType;
(function (AttachLoadBalancerTargetGroupsResultType) {
    AttachLoadBalancerTargetGroupsResultType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttachLoadBalancerTargetGroupsResultType || (AttachLoadBalancerTargetGroupsResultType = {}));
export var AttachLoadBalancerTargetGroupsType;
(function (AttachLoadBalancerTargetGroupsType) {
    AttachLoadBalancerTargetGroupsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttachLoadBalancerTargetGroupsType || (AttachLoadBalancerTargetGroupsType = {}));
export var FailedScheduledUpdateGroupActionRequest;
(function (FailedScheduledUpdateGroupActionRequest) {
    FailedScheduledUpdateGroupActionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FailedScheduledUpdateGroupActionRequest || (FailedScheduledUpdateGroupActionRequest = {}));
export var BatchDeleteScheduledActionAnswer;
(function (BatchDeleteScheduledActionAnswer) {
    BatchDeleteScheduledActionAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchDeleteScheduledActionAnswer || (BatchDeleteScheduledActionAnswer = {}));
export var BatchDeleteScheduledActionType;
(function (BatchDeleteScheduledActionType) {
    BatchDeleteScheduledActionType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchDeleteScheduledActionType || (BatchDeleteScheduledActionType = {}));
export var BatchPutScheduledUpdateGroupActionAnswer;
(function (BatchPutScheduledUpdateGroupActionAnswer) {
    BatchPutScheduledUpdateGroupActionAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchPutScheduledUpdateGroupActionAnswer || (BatchPutScheduledUpdateGroupActionAnswer = {}));
export var ScheduledUpdateGroupActionRequest;
(function (ScheduledUpdateGroupActionRequest) {
    ScheduledUpdateGroupActionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScheduledUpdateGroupActionRequest || (ScheduledUpdateGroupActionRequest = {}));
export var BatchPutScheduledUpdateGroupActionType;
(function (BatchPutScheduledUpdateGroupActionType) {
    BatchPutScheduledUpdateGroupActionType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchPutScheduledUpdateGroupActionType || (BatchPutScheduledUpdateGroupActionType = {}));
export var LimitExceededFault;
(function (LimitExceededFault) {
    LimitExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LimitExceededFault || (LimitExceededFault = {}));
export var CancelInstanceRefreshAnswer;
(function (CancelInstanceRefreshAnswer) {
    CancelInstanceRefreshAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelInstanceRefreshAnswer || (CancelInstanceRefreshAnswer = {}));
export var CancelInstanceRefreshType;
(function (CancelInstanceRefreshType) {
    CancelInstanceRefreshType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelInstanceRefreshType || (CancelInstanceRefreshType = {}));
export var CompleteLifecycleActionAnswer;
(function (CompleteLifecycleActionAnswer) {
    CompleteLifecycleActionAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CompleteLifecycleActionAnswer || (CompleteLifecycleActionAnswer = {}));
export var CompleteLifecycleActionType;
(function (CompleteLifecycleActionType) {
    CompleteLifecycleActionType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CompleteLifecycleActionType || (CompleteLifecycleActionType = {}));
export var LaunchTemplateSpecification;
(function (LaunchTemplateSpecification) {
    LaunchTemplateSpecification.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LaunchTemplateSpecification || (LaunchTemplateSpecification = {}));
export var LifecycleHookSpecification;
(function (LifecycleHookSpecification) {
    LifecycleHookSpecification.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LifecycleHookSpecification || (LifecycleHookSpecification = {}));
export var InstancesDistribution;
(function (InstancesDistribution) {
    InstancesDistribution.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstancesDistribution || (InstancesDistribution = {}));
export var LaunchTemplateOverrides;
(function (LaunchTemplateOverrides) {
    LaunchTemplateOverrides.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LaunchTemplateOverrides || (LaunchTemplateOverrides = {}));
export var LaunchTemplate;
(function (LaunchTemplate) {
    LaunchTemplate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LaunchTemplate || (LaunchTemplate = {}));
export var MixedInstancesPolicy;
(function (MixedInstancesPolicy) {
    MixedInstancesPolicy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MixedInstancesPolicy || (MixedInstancesPolicy = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var CreateAutoScalingGroupType;
(function (CreateAutoScalingGroupType) {
    CreateAutoScalingGroupType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAutoScalingGroupType || (CreateAutoScalingGroupType = {}));
export var Ebs;
(function (Ebs) {
    Ebs.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Ebs || (Ebs = {}));
export var BlockDeviceMapping;
(function (BlockDeviceMapping) {
    BlockDeviceMapping.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BlockDeviceMapping || (BlockDeviceMapping = {}));
export var InstanceMonitoring;
(function (InstanceMonitoring) {
    InstanceMonitoring.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceMonitoring || (InstanceMonitoring = {}));
export var InstanceMetadataEndpointState;
(function (InstanceMetadataEndpointState) {
    InstanceMetadataEndpointState["Disabled"] = "disabled";
    InstanceMetadataEndpointState["Enabled"] = "enabled";
})(InstanceMetadataEndpointState || (InstanceMetadataEndpointState = {}));
export var InstanceMetadataHttpTokensState;
(function (InstanceMetadataHttpTokensState) {
    InstanceMetadataHttpTokensState["Optional"] = "optional";
    InstanceMetadataHttpTokensState["Required"] = "required";
})(InstanceMetadataHttpTokensState || (InstanceMetadataHttpTokensState = {}));
export var InstanceMetadataOptions;
(function (InstanceMetadataOptions) {
    InstanceMetadataOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceMetadataOptions || (InstanceMetadataOptions = {}));
export var CreateLaunchConfigurationType;
(function (CreateLaunchConfigurationType) {
    CreateLaunchConfigurationType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLaunchConfigurationType || (CreateLaunchConfigurationType = {}));
export var CreateOrUpdateTagsType;
(function (CreateOrUpdateTagsType) {
    CreateOrUpdateTagsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateOrUpdateTagsType || (CreateOrUpdateTagsType = {}));
export var ResourceInUseFault;
(function (ResourceInUseFault) {
    ResourceInUseFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceInUseFault || (ResourceInUseFault = {}));
export var DeleteAutoScalingGroupType;
(function (DeleteAutoScalingGroupType) {
    DeleteAutoScalingGroupType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteAutoScalingGroupType || (DeleteAutoScalingGroupType = {}));
export var ScalingActivityInProgressFault;
(function (ScalingActivityInProgressFault) {
    ScalingActivityInProgressFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScalingActivityInProgressFault || (ScalingActivityInProgressFault = {}));
export var LaunchConfigurationNameType;
(function (LaunchConfigurationNameType) {
    LaunchConfigurationNameType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LaunchConfigurationNameType || (LaunchConfigurationNameType = {}));
export var DeleteLifecycleHookAnswer;
(function (DeleteLifecycleHookAnswer) {
    DeleteLifecycleHookAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLifecycleHookAnswer || (DeleteLifecycleHookAnswer = {}));
export var DeleteLifecycleHookType;
(function (DeleteLifecycleHookType) {
    DeleteLifecycleHookType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLifecycleHookType || (DeleteLifecycleHookType = {}));
export var DeleteNotificationConfigurationType;
(function (DeleteNotificationConfigurationType) {
    DeleteNotificationConfigurationType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteNotificationConfigurationType || (DeleteNotificationConfigurationType = {}));
export var DeletePolicyType;
(function (DeletePolicyType) {
    DeletePolicyType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeletePolicyType || (DeletePolicyType = {}));
export var DeleteScheduledActionType;
(function (DeleteScheduledActionType) {
    DeleteScheduledActionType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteScheduledActionType || (DeleteScheduledActionType = {}));
export var DeleteTagsType;
(function (DeleteTagsType) {
    DeleteTagsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteTagsType || (DeleteTagsType = {}));
export var DescribeAccountLimitsAnswer;
(function (DescribeAccountLimitsAnswer) {
    DescribeAccountLimitsAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccountLimitsAnswer || (DescribeAccountLimitsAnswer = {}));
export var DescribeAdjustmentTypesAnswer;
(function (DescribeAdjustmentTypesAnswer) {
    DescribeAdjustmentTypesAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAdjustmentTypesAnswer || (DescribeAdjustmentTypesAnswer = {}));
export var AutoScalingGroupNamesType;
(function (AutoScalingGroupNamesType) {
    AutoScalingGroupNamesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingGroupNamesType || (AutoScalingGroupNamesType = {}));
export var EnabledMetric;
(function (EnabledMetric) {
    EnabledMetric.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnabledMetric || (EnabledMetric = {}));
export var LifecycleState;
(function (LifecycleState) {
    LifecycleState["DETACHED"] = "Detached";
    LifecycleState["DETACHING"] = "Detaching";
    LifecycleState["ENTERING_STANDBY"] = "EnteringStandby";
    LifecycleState["IN_SERVICE"] = "InService";
    LifecycleState["PENDING"] = "Pending";
    LifecycleState["PENDING_PROCEED"] = "Pending:Proceed";
    LifecycleState["PENDING_WAIT"] = "Pending:Wait";
    LifecycleState["QUARANTINED"] = "Quarantined";
    LifecycleState["STANDBY"] = "Standby";
    LifecycleState["TERMINATED"] = "Terminated";
    LifecycleState["TERMINATING"] = "Terminating";
    LifecycleState["TERMINATING_PROCEED"] = "Terminating:Proceed";
    LifecycleState["TERMINATING_WAIT"] = "Terminating:Wait";
})(LifecycleState || (LifecycleState = {}));
export var Instance;
(function (Instance) {
    Instance.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Instance || (Instance = {}));
export var SuspendedProcess;
(function (SuspendedProcess) {
    SuspendedProcess.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SuspendedProcess || (SuspendedProcess = {}));
export var TagDescription;
(function (TagDescription) {
    TagDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagDescription || (TagDescription = {}));
export var AutoScalingGroup;
(function (AutoScalingGroup) {
    AutoScalingGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingGroup || (AutoScalingGroup = {}));
export var AutoScalingGroupsType;
(function (AutoScalingGroupsType) {
    AutoScalingGroupsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingGroupsType || (AutoScalingGroupsType = {}));
export var InvalidNextToken;
(function (InvalidNextToken) {
    InvalidNextToken.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidNextToken || (InvalidNextToken = {}));
export var AutoScalingInstanceDetails;
(function (AutoScalingInstanceDetails) {
    AutoScalingInstanceDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingInstanceDetails || (AutoScalingInstanceDetails = {}));
export var AutoScalingInstancesType;
(function (AutoScalingInstancesType) {
    AutoScalingInstancesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingInstancesType || (AutoScalingInstancesType = {}));
export var DescribeAutoScalingInstancesType;
(function (DescribeAutoScalingInstancesType) {
    DescribeAutoScalingInstancesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAutoScalingInstancesType || (DescribeAutoScalingInstancesType = {}));
export var DescribeAutoScalingNotificationTypesAnswer;
(function (DescribeAutoScalingNotificationTypesAnswer) {
    DescribeAutoScalingNotificationTypesAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAutoScalingNotificationTypesAnswer || (DescribeAutoScalingNotificationTypesAnswer = {}));
export var InstanceRefreshStatus;
(function (InstanceRefreshStatus) {
    InstanceRefreshStatus["Cancelled"] = "Cancelled";
    InstanceRefreshStatus["Cancelling"] = "Cancelling";
    InstanceRefreshStatus["Failed"] = "Failed";
    InstanceRefreshStatus["InProgress"] = "InProgress";
    InstanceRefreshStatus["Pending"] = "Pending";
    InstanceRefreshStatus["Successful"] = "Successful";
})(InstanceRefreshStatus || (InstanceRefreshStatus = {}));
export var InstanceRefresh;
(function (InstanceRefresh) {
    InstanceRefresh.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceRefresh || (InstanceRefresh = {}));
export var DescribeInstanceRefreshesAnswer;
(function (DescribeInstanceRefreshesAnswer) {
    DescribeInstanceRefreshesAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceRefreshesAnswer || (DescribeInstanceRefreshesAnswer = {}));
export var DescribeInstanceRefreshesType;
(function (DescribeInstanceRefreshesType) {
    DescribeInstanceRefreshesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceRefreshesType || (DescribeInstanceRefreshesType = {}));
export var LaunchConfigurationNamesType;
(function (LaunchConfigurationNamesType) {
    LaunchConfigurationNamesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LaunchConfigurationNamesType || (LaunchConfigurationNamesType = {}));
export var LaunchConfiguration;
(function (LaunchConfiguration) {
    LaunchConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LaunchConfiguration || (LaunchConfiguration = {}));
export var LaunchConfigurationsType;
(function (LaunchConfigurationsType) {
    LaunchConfigurationsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LaunchConfigurationsType || (LaunchConfigurationsType = {}));
export var LifecycleHook;
(function (LifecycleHook) {
    LifecycleHook.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LifecycleHook || (LifecycleHook = {}));
export var DescribeLifecycleHooksAnswer;
(function (DescribeLifecycleHooksAnswer) {
    DescribeLifecycleHooksAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLifecycleHooksAnswer || (DescribeLifecycleHooksAnswer = {}));
export var DescribeLifecycleHooksType;
(function (DescribeLifecycleHooksType) {
    DescribeLifecycleHooksType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLifecycleHooksType || (DescribeLifecycleHooksType = {}));
export var DescribeLifecycleHookTypesAnswer;
(function (DescribeLifecycleHookTypesAnswer) {
    DescribeLifecycleHookTypesAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLifecycleHookTypesAnswer || (DescribeLifecycleHookTypesAnswer = {}));
export var DescribeLoadBalancersRequest;
(function (DescribeLoadBalancersRequest) {
    DescribeLoadBalancersRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancersRequest || (DescribeLoadBalancersRequest = {}));
export var LoadBalancerState;
(function (LoadBalancerState) {
    LoadBalancerState.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LoadBalancerState || (LoadBalancerState = {}));
export var DescribeLoadBalancersResponse;
(function (DescribeLoadBalancersResponse) {
    DescribeLoadBalancersResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancersResponse || (DescribeLoadBalancersResponse = {}));
export var DescribeLoadBalancerTargetGroupsRequest;
(function (DescribeLoadBalancerTargetGroupsRequest) {
    DescribeLoadBalancerTargetGroupsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancerTargetGroupsRequest || (DescribeLoadBalancerTargetGroupsRequest = {}));
export var LoadBalancerTargetGroupState;
(function (LoadBalancerTargetGroupState) {
    LoadBalancerTargetGroupState.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LoadBalancerTargetGroupState || (LoadBalancerTargetGroupState = {}));
export var DescribeLoadBalancerTargetGroupsResponse;
(function (DescribeLoadBalancerTargetGroupsResponse) {
    DescribeLoadBalancerTargetGroupsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancerTargetGroupsResponse || (DescribeLoadBalancerTargetGroupsResponse = {}));
export var MetricGranularityType;
(function (MetricGranularityType) {
    MetricGranularityType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricGranularityType || (MetricGranularityType = {}));
export var MetricCollectionType;
(function (MetricCollectionType) {
    MetricCollectionType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricCollectionType || (MetricCollectionType = {}));
export var DescribeMetricCollectionTypesAnswer;
(function (DescribeMetricCollectionTypesAnswer) {
    DescribeMetricCollectionTypesAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeMetricCollectionTypesAnswer || (DescribeMetricCollectionTypesAnswer = {}));
export var NotificationConfiguration;
(function (NotificationConfiguration) {
    NotificationConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NotificationConfiguration || (NotificationConfiguration = {}));
export var DescribeNotificationConfigurationsAnswer;
(function (DescribeNotificationConfigurationsAnswer) {
    DescribeNotificationConfigurationsAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeNotificationConfigurationsAnswer || (DescribeNotificationConfigurationsAnswer = {}));
export var DescribeNotificationConfigurationsType;
(function (DescribeNotificationConfigurationsType) {
    DescribeNotificationConfigurationsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeNotificationConfigurationsType || (DescribeNotificationConfigurationsType = {}));
export var DescribePoliciesType;
(function (DescribePoliciesType) {
    DescribePoliciesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribePoliciesType || (DescribePoliciesType = {}));
export var StepAdjustment;
(function (StepAdjustment) {
    StepAdjustment.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StepAdjustment || (StepAdjustment = {}));
export var MetricDimension;
(function (MetricDimension) {
    MetricDimension.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricDimension || (MetricDimension = {}));
export var MetricStatistic;
(function (MetricStatistic) {
    MetricStatistic["Average"] = "Average";
    MetricStatistic["Maximum"] = "Maximum";
    MetricStatistic["Minimum"] = "Minimum";
    MetricStatistic["SampleCount"] = "SampleCount";
    MetricStatistic["Sum"] = "Sum";
})(MetricStatistic || (MetricStatistic = {}));
export var CustomizedMetricSpecification;
(function (CustomizedMetricSpecification) {
    CustomizedMetricSpecification.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomizedMetricSpecification || (CustomizedMetricSpecification = {}));
export var MetricType;
(function (MetricType) {
    MetricType["ALBRequestCountPerTarget"] = "ALBRequestCountPerTarget";
    MetricType["ASGAverageCPUUtilization"] = "ASGAverageCPUUtilization";
    MetricType["ASGAverageNetworkIn"] = "ASGAverageNetworkIn";
    MetricType["ASGAverageNetworkOut"] = "ASGAverageNetworkOut";
})(MetricType || (MetricType = {}));
export var PredefinedMetricSpecification;
(function (PredefinedMetricSpecification) {
    PredefinedMetricSpecification.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PredefinedMetricSpecification || (PredefinedMetricSpecification = {}));
export var TargetTrackingConfiguration;
(function (TargetTrackingConfiguration) {
    TargetTrackingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TargetTrackingConfiguration || (TargetTrackingConfiguration = {}));
export var ScalingPolicy;
(function (ScalingPolicy) {
    ScalingPolicy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScalingPolicy || (ScalingPolicy = {}));
export var PoliciesType;
(function (PoliciesType) {
    PoliciesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PoliciesType || (PoliciesType = {}));
export var DescribeScalingActivitiesType;
(function (DescribeScalingActivitiesType) {
    DescribeScalingActivitiesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeScalingActivitiesType || (DescribeScalingActivitiesType = {}));
export var ProcessType;
(function (ProcessType) {
    ProcessType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProcessType || (ProcessType = {}));
export var ProcessesType;
(function (ProcessesType) {
    ProcessesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProcessesType || (ProcessesType = {}));
export var DescribeScheduledActionsType;
(function (DescribeScheduledActionsType) {
    DescribeScheduledActionsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeScheduledActionsType || (DescribeScheduledActionsType = {}));
export var ScheduledUpdateGroupAction;
(function (ScheduledUpdateGroupAction) {
    ScheduledUpdateGroupAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScheduledUpdateGroupAction || (ScheduledUpdateGroupAction = {}));
export var ScheduledActionsType;
(function (ScheduledActionsType) {
    ScheduledActionsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScheduledActionsType || (ScheduledActionsType = {}));
export var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Filter || (Filter = {}));
export var DescribeTagsType;
(function (DescribeTagsType) {
    DescribeTagsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTagsType || (DescribeTagsType = {}));
export var TagsType;
(function (TagsType) {
    TagsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagsType || (TagsType = {}));
export var DescribeTerminationPolicyTypesAnswer;
(function (DescribeTerminationPolicyTypesAnswer) {
    DescribeTerminationPolicyTypesAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTerminationPolicyTypesAnswer || (DescribeTerminationPolicyTypesAnswer = {}));
export var DetachInstancesAnswer;
(function (DetachInstancesAnswer) {
    DetachInstancesAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachInstancesAnswer || (DetachInstancesAnswer = {}));
export var DetachInstancesQuery;
(function (DetachInstancesQuery) {
    DetachInstancesQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachInstancesQuery || (DetachInstancesQuery = {}));
export var DetachLoadBalancersResultType;
(function (DetachLoadBalancersResultType) {
    DetachLoadBalancersResultType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachLoadBalancersResultType || (DetachLoadBalancersResultType = {}));
export var DetachLoadBalancersType;
(function (DetachLoadBalancersType) {
    DetachLoadBalancersType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachLoadBalancersType || (DetachLoadBalancersType = {}));
export var DetachLoadBalancerTargetGroupsResultType;
(function (DetachLoadBalancerTargetGroupsResultType) {
    DetachLoadBalancerTargetGroupsResultType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachLoadBalancerTargetGroupsResultType || (DetachLoadBalancerTargetGroupsResultType = {}));
export var DetachLoadBalancerTargetGroupsType;
(function (DetachLoadBalancerTargetGroupsType) {
    DetachLoadBalancerTargetGroupsType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachLoadBalancerTargetGroupsType || (DetachLoadBalancerTargetGroupsType = {}));
export var DisableMetricsCollectionQuery;
(function (DisableMetricsCollectionQuery) {
    DisableMetricsCollectionQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisableMetricsCollectionQuery || (DisableMetricsCollectionQuery = {}));
export var EnableMetricsCollectionQuery;
(function (EnableMetricsCollectionQuery) {
    EnableMetricsCollectionQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnableMetricsCollectionQuery || (EnableMetricsCollectionQuery = {}));
export var EnterStandbyAnswer;
(function (EnterStandbyAnswer) {
    EnterStandbyAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnterStandbyAnswer || (EnterStandbyAnswer = {}));
export var EnterStandbyQuery;
(function (EnterStandbyQuery) {
    EnterStandbyQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnterStandbyQuery || (EnterStandbyQuery = {}));
export var ExecutePolicyType;
(function (ExecutePolicyType) {
    ExecutePolicyType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExecutePolicyType || (ExecutePolicyType = {}));
export var ExitStandbyAnswer;
(function (ExitStandbyAnswer) {
    ExitStandbyAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExitStandbyAnswer || (ExitStandbyAnswer = {}));
export var ExitStandbyQuery;
(function (ExitStandbyQuery) {
    ExitStandbyQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExitStandbyQuery || (ExitStandbyQuery = {}));
export var PutLifecycleHookAnswer;
(function (PutLifecycleHookAnswer) {
    PutLifecycleHookAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutLifecycleHookAnswer || (PutLifecycleHookAnswer = {}));
export var PutLifecycleHookType;
(function (PutLifecycleHookType) {
    PutLifecycleHookType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutLifecycleHookType || (PutLifecycleHookType = {}));
export var PutNotificationConfigurationType;
(function (PutNotificationConfigurationType) {
    PutNotificationConfigurationType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutNotificationConfigurationType || (PutNotificationConfigurationType = {}));
export var PolicyARNType;
(function (PolicyARNType) {
    PolicyARNType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyARNType || (PolicyARNType = {}));
export var PutScalingPolicyType;
(function (PutScalingPolicyType) {
    PutScalingPolicyType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutScalingPolicyType || (PutScalingPolicyType = {}));
export var PutScheduledUpdateGroupActionType;
(function (PutScheduledUpdateGroupActionType) {
    PutScheduledUpdateGroupActionType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutScheduledUpdateGroupActionType || (PutScheduledUpdateGroupActionType = {}));
export var RecordLifecycleActionHeartbeatAnswer;
(function (RecordLifecycleActionHeartbeatAnswer) {
    RecordLifecycleActionHeartbeatAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RecordLifecycleActionHeartbeatAnswer || (RecordLifecycleActionHeartbeatAnswer = {}));
export var RecordLifecycleActionHeartbeatType;
(function (RecordLifecycleActionHeartbeatType) {
    RecordLifecycleActionHeartbeatType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RecordLifecycleActionHeartbeatType || (RecordLifecycleActionHeartbeatType = {}));
export var ScalingProcessQuery;
(function (ScalingProcessQuery) {
    ScalingProcessQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScalingProcessQuery || (ScalingProcessQuery = {}));
export var SetDesiredCapacityType;
(function (SetDesiredCapacityType) {
    SetDesiredCapacityType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetDesiredCapacityType || (SetDesiredCapacityType = {}));
export var SetInstanceHealthQuery;
(function (SetInstanceHealthQuery) {
    SetInstanceHealthQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetInstanceHealthQuery || (SetInstanceHealthQuery = {}));
export var SetInstanceProtectionAnswer;
(function (SetInstanceProtectionAnswer) {
    SetInstanceProtectionAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetInstanceProtectionAnswer || (SetInstanceProtectionAnswer = {}));
export var SetInstanceProtectionQuery;
(function (SetInstanceProtectionQuery) {
    SetInstanceProtectionQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetInstanceProtectionQuery || (SetInstanceProtectionQuery = {}));
export var InstanceRefreshInProgressFault;
(function (InstanceRefreshInProgressFault) {
    InstanceRefreshInProgressFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceRefreshInProgressFault || (InstanceRefreshInProgressFault = {}));
export var StartInstanceRefreshAnswer;
(function (StartInstanceRefreshAnswer) {
    StartInstanceRefreshAnswer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartInstanceRefreshAnswer || (StartInstanceRefreshAnswer = {}));
export var RefreshPreferences;
(function (RefreshPreferences) {
    RefreshPreferences.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RefreshPreferences || (RefreshPreferences = {}));
export var RefreshStrategy;
(function (RefreshStrategy) {
    RefreshStrategy["Rolling"] = "Rolling";
})(RefreshStrategy || (RefreshStrategy = {}));
export var StartInstanceRefreshType;
(function (StartInstanceRefreshType) {
    StartInstanceRefreshType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartInstanceRefreshType || (StartInstanceRefreshType = {}));
export var TerminateInstanceInAutoScalingGroupType;
(function (TerminateInstanceInAutoScalingGroupType) {
    TerminateInstanceInAutoScalingGroupType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TerminateInstanceInAutoScalingGroupType || (TerminateInstanceInAutoScalingGroupType = {}));
export var UpdateAutoScalingGroupType;
(function (UpdateAutoScalingGroupType) {
    UpdateAutoScalingGroupType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateAutoScalingGroupType || (UpdateAutoScalingGroupType = {}));
