// @aws-sdk/client-auto-scaling/dist/cjs/commands/CancelInstanceRefreshCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CancelInstanceRefreshCommand = void 0;
const models_0_1 = require("../models/models_0");
const Aws_query_1 = require("../protocols/Aws_query");
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
/**
 * <p>Cancels an instance refresh operation in progress. Cancellation does not roll back any
 *             replacements that have already been completed, but it prevents new replacements from
 *             being started. </p>
 *         <p>For more information, see <a href="https://docs.aws.amazon.com/autoscaling/ec2/userguide/asg-instance-refresh.html">Replacing Auto Scaling Instances
 *                 Based on an Instance Refresh</a>.</p>
 */
class CancelInstanceRefreshCommand extends smithy_client_1.Command {
    // Start section: command_properties
    // End section: command_properties
    constructor(input) {
        // Start section: command_constructor
        super();
        this.input = input;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "AutoScalingClient";
        const commandName = "CancelInstanceRefreshCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_0_1.CancelInstanceRefreshType.filterSensitiveLog,
            outputFilterSensitiveLog: models_0_1.CancelInstanceRefreshAnswer.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_query_1.serializeAws_queryCancelInstanceRefreshCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_query_1.deserializeAws_queryCancelInstanceRefreshCommand(output, context);
    }
}
exports.CancelInstanceRefreshCommand = CancelInstanceRefreshCommand;
