// @aws-sdk/client-auto-scaling/dist/cjs/commands/DescribeScheduledActionsCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeScheduledActionsCommand = void 0;
const models_0_1 = require("../models/models_0");
const Aws_query_1 = require("../protocols/Aws_query");
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
/**
 * <p>Describes the actions scheduled for your Auto Scaling group that haven't run or that have not
 *             reached their end time. To describe the actions that have already run, call the <a>DescribeScalingActivities</a> API.</p>
 */
class DescribeScheduledActionsCommand extends smithy_client_1.Command {
    // Start section: command_properties
    // End section: command_properties
    constructor(input) {
        // Start section: command_constructor
        super();
        this.input = input;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "AutoScalingClient";
        const commandName = "DescribeScheduledActionsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_0_1.DescribeScheduledActionsType.filterSensitiveLog,
            outputFilterSensitiveLog: models_0_1.ScheduledActionsType.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_query_1.serializeAws_queryDescribeScheduledActionsCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_query_1.deserializeAws_queryDescribeScheduledActionsCommand(output, context);
    }
}
exports.DescribeScheduledActionsCommand = DescribeScheduledActionsCommand;
