// @aws-sdk/client-auto-scaling/dist/es/commands/DescribeScalingActivitiesCommand.js
import { __extends } from "tslib";
import { ActivitiesType, DescribeScalingActivitiesType } from "../models/models_0";
import { deserializeAws_queryDescribeScalingActivitiesCommand, serializeAws_queryDescribeScalingActivitiesCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Describes one or more scaling activities for the specified Auto Scaling group.</p>
 */
var DescribeScalingActivitiesCommand = /** @class */ (function (_super) {
    __extends(DescribeScalingActivitiesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeScalingActivitiesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeScalingActivitiesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AutoScalingClient";
        var commandName = "DescribeScalingActivitiesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeScalingActivitiesType.filterSensitiveLog,
            outputFilterSensitiveLog: ActivitiesType.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeScalingActivitiesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeScalingActivitiesCommand(input, context);
    };
    DescribeScalingActivitiesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeScalingActivitiesCommand(output, context);
    };
    return DescribeScalingActivitiesCommand;
}($Command));
export { DescribeScalingActivitiesCommand };
