// @aws-sdk/client-auto-scaling/dist/es/commands/DescribePoliciesCommand.js
import { __extends } from "tslib";
import { DescribePoliciesType, PoliciesType } from "../models/models_0";
import { deserializeAws_queryDescribePoliciesCommand, serializeAws_queryDescribePoliciesCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Describes the policies for the specified Auto Scaling group.</p>
 */
var DescribePoliciesCommand = /** @class */ (function (_super) {
    __extends(DescribePoliciesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribePoliciesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribePoliciesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AutoScalingClient";
        var commandName = "DescribePoliciesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribePoliciesType.filterSensitiveLog,
            outputFilterSensitiveLog: PoliciesType.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribePoliciesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribePoliciesCommand(input, context);
    };
    DescribePoliciesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribePoliciesCommand(output, context);
    };
    return DescribePoliciesCommand;
}($Command));
export { DescribePoliciesCommand };
