// @aws-sdk/client-auto-scaling/dist/es/commands/DescribeAutoScalingGroupsCommand.js
import { __extends } from "tslib";
import { AutoScalingGroupNamesType, AutoScalingGroupsType } from "../models/models_0";
import { deserializeAws_queryDescribeAutoScalingGroupsCommand, serializeAws_queryDescribeAutoScalingGroupsCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Describes one or more Auto Scaling groups.</p>
 */
var DescribeAutoScalingGroupsCommand = /** @class */ (function (_super) {
    __extends(DescribeAutoScalingGroupsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeAutoScalingGroupsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeAutoScalingGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AutoScalingClient";
        var commandName = "DescribeAutoScalingGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AutoScalingGroupNamesType.filterSensitiveLog,
            outputFilterSensitiveLog: AutoScalingGroupsType.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeAutoScalingGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeAutoScalingGroupsCommand(input, context);
    };
    DescribeAutoScalingGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeAutoScalingGroupsCommand(output, context);
    };
    return DescribeAutoScalingGroupsCommand;
}($Command));
export { DescribeAutoScalingGroupsCommand };
