// @aws-sdk/client-auto-scaling/dist/es/commands/DescribeLoadBalancerTargetGroupsCommand.js
import { __extends } from "tslib";
import { DescribeLoadBalancerTargetGroupsRequest, DescribeLoadBalancerTargetGroupsResponse } from "../models/models_0";
import { deserializeAws_queryDescribeLoadBalancerTargetGroupsCommand, serializeAws_queryDescribeLoadBalancerTargetGroupsCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Describes the target groups for the specified Auto Scaling group.</p>
 */
var DescribeLoadBalancerTargetGroupsCommand = /** @class */ (function (_super) {
    __extends(DescribeLoadBalancerTargetGroupsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeLoadBalancerTargetGroupsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeLoadBalancerTargetGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AutoScalingClient";
        var commandName = "DescribeLoadBalancerTargetGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeLoadBalancerTargetGroupsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeLoadBalancerTargetGroupsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeLoadBalancerTargetGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeLoadBalancerTargetGroupsCommand(input, context);
    };
    DescribeLoadBalancerTargetGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeLoadBalancerTargetGroupsCommand(output, context);
    };
    return DescribeLoadBalancerTargetGroupsCommand;
}($Command));
export { DescribeLoadBalancerTargetGroupsCommand };
