// @aws-sdk/client-auto-scaling/dist/es/commands/DescribeLifecycleHookTypesCommand.js
import { __extends } from "tslib";
import { DescribeLifecycleHookTypesAnswer } from "../models/models_0";
import { deserializeAws_queryDescribeLifecycleHookTypesCommand, serializeAws_queryDescribeLifecycleHookTypesCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Describes the available types of lifecycle hooks.</p>
 *         <p>The following hook types are supported:</p>
 *         <ul>
 *             <li>
 *                 <p>autoscaling:EC2_INSTANCE_LAUNCHING</p>
 *             </li>
 *             <li>
 *                 <p>autoscaling:EC2_INSTANCE_TERMINATING</p>
 *             </li>
 *          </ul>
 */
var DescribeLifecycleHookTypesCommand = /** @class */ (function (_super) {
    __extends(DescribeLifecycleHookTypesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeLifecycleHookTypesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeLifecycleHookTypesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AutoScalingClient";
        var commandName = "DescribeLifecycleHookTypesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: function (input) { return input; },
            outputFilterSensitiveLog: DescribeLifecycleHookTypesAnswer.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeLifecycleHookTypesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeLifecycleHookTypesCommand(input, context);
    };
    DescribeLifecycleHookTypesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeLifecycleHookTypesCommand(output, context);
    };
    return DescribeLifecycleHookTypesCommand;
}($Command));
export { DescribeLifecycleHookTypesCommand };
