// @aws-sdk/client-auto-scaling/dist/es/commands/SetInstanceHealthCommand.js
import { __extends } from "tslib";
import { SetInstanceHealthQuery } from "../models/models_0";
import { deserializeAws_querySetInstanceHealthCommand, serializeAws_querySetInstanceHealthCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Sets the health status of the specified instance.</p>
 *         <p>For more information, see <a href="https://docs.aws.amazon.com/autoscaling/ec2/userguide/healthcheck.html">Health checks for Auto Scaling
 *                 instances</a> in the <i>Amazon EC2 Auto Scaling User Guide</i>.</p>
 */
var SetInstanceHealthCommand = /** @class */ (function (_super) {
    __extends(SetInstanceHealthCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function SetInstanceHealthCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    SetInstanceHealthCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AutoScalingClient";
        var commandName = "SetInstanceHealthCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetInstanceHealthQuery.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetInstanceHealthCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetInstanceHealthCommand(input, context);
    };
    SetInstanceHealthCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetInstanceHealthCommand(output, context);
    };
    return SetInstanceHealthCommand;
}($Command));
export { SetInstanceHealthCommand };
