// @aws-sdk/client-ecs/dist/es/commands/UpdateContainerAgentCommand.js
import { __extends } from "tslib";
import { UpdateContainerAgentRequest, UpdateContainerAgentResponse } from "../models/models_0";
import { deserializeAws_json1_1UpdateContainerAgentCommand, serializeAws_json1_1UpdateContainerAgentCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Updates the Amazon ECS container agent on a specified container instance. Updating the
 * 			Amazon ECS container agent does not interrupt running tasks or services on the container
 * 			instance. The process for updating the agent differs depending on whether your container
 * 			instance was launched with the Amazon ECS-optimized AMI or another operating system.</p>
 * 		       <p>
 * 			         <code>UpdateContainerAgent</code> requires the Amazon ECS-optimized AMI or Amazon Linux with
 * 			the <code>ecs-init</code> service installed and running. For help updating the Amazon ECS
 * 			container agent on other operating systems, see <a href="https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-update.html#manually_update_agent">Manually Updating the Amazon ECS Container Agent</a> in the
 * 				<i>Amazon Elastic Container Service Developer Guide</i>.</p>
 */
var UpdateContainerAgentCommand = /** @class */ (function (_super) {
    __extends(UpdateContainerAgentCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function UpdateContainerAgentCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    UpdateContainerAgentCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECSClient";
        var commandName = "UpdateContainerAgentCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateContainerAgentRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateContainerAgentResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateContainerAgentCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateContainerAgentCommand(input, context);
    };
    UpdateContainerAgentCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateContainerAgentCommand(output, context);
    };
    return UpdateContainerAgentCommand;
}($Command));
export { UpdateContainerAgentCommand };
